wmf/handlers.go

```diff
@@ -536,18 +429,31 @@ func (self *Handler) verifyFxAAssertion(assertion string) (userid, email string,
 	// the response has either been a redirect or a validated assertion.
 	// fun times, fun times...
 
+	verr := "no idpClaims"
+
 	if idp, ok := buff["idpClaims"]; ok {
+		// It's a validated Assertion, so get the userid and email.
+		verr = "no principle"
 		if principal, ok := idp.(map[string]interface{})["principal"]; ok {
+			verr = "no email"
 			if uid, ok := principal.(map[string]interface{})["email"]; ok {
+				verr = "bad email str"
 				userid = strings.Split(uid.(string), "@")[0]
 			}
 		}
-		if email, ok := idp.(map[string]interface{})["fxa-verifiedEmail"]; ok {
-			if userid == "" {
-				userid = self.genHash(email.(string))
-			}
-			return userid, email.(string), nil
+
+		if iemail, ok := idp.(map[string]interface{})["fxa-verifiedEmail"]; ok {
```

Comment:
Could there ever be an assertion with a user ID (`principal.email`), but without an `fxa-verifiedEmail`?